src/order/bounds.lean

```diff
@@ -1,378 +1,609 @@
+@is_glb_empty (order_dual γ) _
 
-lemma bdd_below_iff_subset_Ici (s : set α) : bdd_below s ↔ ∃ a, s ⊆ Ici a := iff.rfl
+lemma is_lub.nonempty [no_bot_order α] (hs : is_lub s a) : s.nonempty :=
+let ⟨a', ha'⟩ := no_bot a in
+ne_empty_iff_nonempty.1 $ assume h,
+have a ≤ a', from hs.right $ by simp only [h, upper_bounds_empty],
+not_le_of_lt ha' this
+
+lemma is_glb.nonempty [no_top_order α] (hs : is_glb s a) : s.nonempty :=
+@is_lub.nonempty (order_dual α) _ _ _ _ hs
 
-lemma bdd_above_iff_subset_Iic (s : set α) : bdd_above s ↔ ∃ a, s ⊆ Iic a := iff.rfl
+/-!
+### insert
+-/
 
-lemma bdd_below_bdd_above_iff_subset_Icc (s : set α) :
-  bdd_below s ∧ bdd_above s ↔ ∃ a b, s ⊆ Icc a b :=
-by simp [Ici_inter_Iic.symm, subset_inter_iff, bdd_below_iff_subset_Ici, bdd_above_iff_subset_Iic]
+/-- Adding a point to a set preserves its boundedness above. -/
+@[simp] lemma bdd_above_insert [semilattice_sup γ] (a : γ) {s : set γ} :
+  bdd_above (insert a s) ↔ bdd_above s :=
+by simp only [insert_eq, bdd_above_union, bdd_above_singleton, true_and]
 
-end preorder
+lemma bdd_above.insert [semilattice_sup γ] (a : γ) {s : set γ} (hs : bdd_above s) :
+  bdd_above (insert a s) :=
+(bdd_above_insert a).2 hs
+
+/--Adding a point to a set preserves its boundedness below.-/
+@[simp] lemma bdd_below_insert [semilattice_inf γ] (a : γ) {s : set γ} :
+  bdd_below (insert a s) ↔ bdd_below s :=
+by simp only [insert_eq, bdd_below_union, bdd_below_singleton, true_and]
+
+lemma bdd_below.insert [semilattice_inf γ] (a : γ) {s : set γ} (hs : bdd_below s) :
+  bdd_below (insert a s) :=
+(bdd_below_insert a).2 hs
+
+lemma is_lub.insert [semilattice_sup γ] (a) {b} {s : set γ} (hs : is_lub s b) :
+  is_lub (insert a s) (a ⊔ b) :=
+by { rw insert_eq, exact is_lub_singleton.union hs }
+
+lemma is_glb.insert [semilattice_inf γ] (a) {b} {s : set γ} (hs : is_glb s b) :
+  is_glb (insert a s) (a ⊓ b) :=
+by { rw insert_eq, exact is_glb_singleton.union hs }
+
+lemma is_greatest.insert [decidable_linear_order γ] (a) {b} {s : set γ} (hs : is_greatest s b) :
+  is_greatest (insert a s) (max a b) :=
+by { rw insert_eq, exact is_greatest_singleton.union hs }
 
-/--When there is a global maximum, every set is bounded above.-/
-@[simp] lemma bdd_above_top [order_top α] (s : set α) : bdd_above s :=
+lemma is_least.insert [decidable_linear_order γ] (a) {b} {s : set γ} (hs : is_least s b) :
+  is_least (insert a s) (min a b) :=
+by { rw insert_eq, exact is_least_singleton.union hs }
+
+lemma upper_bounds_insert (a : α) (s : set α) :
+  upper_bounds (insert a s) = Ici a ∩ upper_bounds s :=
+by rw [insert_eq, upper_bounds_union, upper_bounds_singleton]
+
+lemma lower_bounds_insert (a : α) (s : set α) :
+  lower_bounds (insert a s) = Iic a ∩ lower_bounds s :=
+by rw [insert_eq, lower_bounds_union, lower_bounds_singleton]
+
+/-- When there is a global maximum, every set is bounded above. -/
+@[simp] lemma bdd_above_top [order_top γ] (s : set γ) : bdd_above s :=
```

Comment:
I know this is the preexisting name in the library, but since you are refactoring everything you might as well change it to something better like `order_top.bdd_above`. Same thing for `bdd_below_bot`.

Reply:
Done.